scipy/stats/mstats_basic.py

```diff
@@ -2297,14 +2302,15 @@ def skew(a, axis=0, bias=True):
 
     """
     a, axis = _chk_asarray(a,axis)
-    n = a.count(axis)
-    m2 = moment(a, 2, axis)
-    m3 = moment(a, 3, axis)
-    zero = (m2 <= (np.finfo(m2.dtype).resolution * a.mean(axis))**2)
+    mean = a.mean(axis, keepdims=True)
+    m2 = _moment(a, 2, axis, mean=mean)
+    m3 = _moment(a, 3, axis, mean=mean)
+    zero = (m2 <= (np.finfo(m2.dtype).resolution * mean.squeeze(axis))**2)
```

Comment:
`_moment` needs `mean` with the extra dimension, but default behavior of `a.mean(axis)` is to eliminate the dimension, so squeeze it.
Could this cause problems if other singleton dimensions are inadvertently removed before comparison with `m2`? Should we be more precise about eliminating only the appropriate axis?

Reply:
>Should we be more precise about eliminating only the appropriate axis?

That's why I pass the `axis` argument. It will only remove the reduction dimensions. e.g.
```python
>>> a = np.random.rand(1, 1, 1, 10)
>>> scipy.stats.kurtosis(a, axis=-1).shape
(1, 1, 1)
```

Reply:
If you're concerned, I could add a test case.

Reply:
Oops - totally missed that somehow. Not concerned. Anything else to be changed or is this ready from your perspective?

Reply:
IMO this is good to go.

Reply:
OK. I'll give others a little while to comment, and if there's nothing else, I'll merge.